### PR TITLE
fix(manager_cli): set --force-non-ssl-session-port if encryption enabled but port not defined

### PR DESCRIPTION
**Pull request by karol-kokoszka**
There is long lasting issue in Scylla https://github.com/scylladb/scylladb/issues/7206#issuecomment-689598343
which causes Scylla to fallback the SSL port to non-SSL one when the encryption on session is enabled, but SSL port is not defined in config.
Manager sanity tests checks if the healthcheck works with encryption enabled, but at the same time, the scylla configuration used for this test, doesn't specify SSL port (so, it fallsback to non-SSL).
See the scylla.yaml from test
```
alternator_enforce_authorization: false
api_address: 127.0.0.1
api_doc_dir: /opt/scylladb/api/api-doc/
api_port: 10000
api_ui_dir: /opt/scylladb/swagger-ui/dist/
audit: table
audit_categories: DCL,DDL,AUTH,ADMIN
auto_bootstrap: true
batch_size_fail_threshold_in_kb: 1024
batch_size_warn_threshold_in_kb: 128
broadcast_rpc_address: 10.15.3.97
cas_contention_timeout_in_ms: 1000
client_encryption_options:
  certificate: /etc/scylla/ssl_conf/client/test.crt
  enabled: true               <---------------------------- ENABLED but native_transport_port_ssl is missing
  keyfile: /etc/scylla/ssl_conf/client/test.key
  truststore: /etc/scylla/ssl_conf/client/catest.pem
cluster_name: manager-regression-manager--db-cluster-53ae9be8
commitlog_segment_size_in_mb: 32
commitlog_sync: periodic
commitlog_sync_period_in_ms: 10000
commitlog_total_space_in_mb: -1
enable_ipv6_dns_lookup: false
endpoint_snitch: org.apache.cassandra.locator.Ec2Snitch
experimental: true
force_schema_commit_log: true
hinted_handoff_enabled: false
listen_address: 10.15.3.97
murmur3_partitioner_ignore_msb_bits: 12
native_shard_aware_transport_port: 19042
native_transport_port: 9042
num_tokens: 256
partitioner: org.apache.cassandra.dht.Murmur3Partitioner
prometheus_address: 0.0.0.0
read_request_timeout_in_ms: 5000
rpc_address: 10.15.3.97
rpc_port: 9160
seed_provider:
- class_name: org.apache.cassandra.locator.SimpleSeedProvider
  parameters:
  - seeds: 10.12.3.185,10.12.1.112,10.15.3.97
write_request_timeout_in_ms: 2000
```

According to the scylla's behavior, manager is expected to fallback to non-SSL port. The point is that it a bad habit on SM side to always try to connect to this SSL port first and then fallback to non-SSL if connection cannot be established. This check is the only way SM could realize that SSL port is not specified.

Starting with manager-3.2.6, CLI supports additional flag/parameter on cluster. It's --force-non-ssl-session-port which defines to always use non-SSL port, even if the encryption is enabled.
The fallback and the check if SSL port is opened is removed from SM 3.2.6.

This PR updates manager test to update cluster with --force-non-ssl-session-port when the encryption is enabled.

SCT sanity fails without it https://jenkins.scylladb.com/view/scylla-manager/job/manager-3.2/job/centos-sanity-test/25
Works well with this change introduced https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/karol-kokoszka/job/manager-3.2-debian-11-sanity/5/

**This is the new PR, where the base is branched out from origin.manager-3.2 and 7187e4d04d03686e0a083d7a1b12c21398afbd5f is cherry-picked from https://github.com/scylladb/scylla-cluster-tests/pull/7231**

### Testing

- [ ]

### PR pre-checks (self review)

- [ ] I added the relevant `backport` labels
- [ ] I didn't leave commented-out/debugging code

### Reminders

- Add New configuration option and document them (in `sdcm/sct_config.py`)
- Add unit tests to cover my changes (under `unit-test/` folder)
- Update the Readme/doc folder relevent to this change (if needed)

### CODE REVIEW

**Comment by karol-kokoszka** [comment]
Test execution against the latest manager (but the guard was change to support --force-non-ssl-session-port with 3.2.5 just for this test run, as 3.2.6 is not tagged yet). 

https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/karol-kokoszka/job/manager-3.2-debian-11-sanity/5/ PASS

**Review by rayakurl — DISMISSED** [review]
LGTM

**Comment by karol-kokoszka** [comment]
> This branch has no conflicts with the base branch
Only those with [write access](https://docs.github.com/articles/what-are-the-different-access-permissions) to this repository can merge pull requests.

@fruch @rayakurl Thanks for approval, but I cannot merge it to manager-3.2 due to permissions lack.
Who can merge it ?
Do you know why I cannot see the details about failed SCT unit-tests ? 
image

Failed workflow 3 -> no page
image

**Comment by fruch** [comment]
@karol-kokoszka 

hooks were failing:

```
12:31:16  commitlint...............................................................Failed
12:31:16  - hook id: commitlint
12:31:16  - exit code: 1
12:31:16  
12:31:16  ⧗   input: fix(manager_cli): set --force-non-ssl-session-port if encryption enabled but port not defined
12:31:16  ✖   header must not be longer than 72 characters, current length is 93 [header-max-length]
12:31:16  
12:31:16  ✖   found 1 problems, 0 warnings
12:31:16  ⓘ   Get help: https://github.com/conventional-changelog/commitlint/#what-is-commitlint
...
12:33:26  sdcm/mgmt/operator.py:388:4: W0221: Number of parameters was 5 in 'ManagerCluster.update' and is now 4 in overridden 'OperatorManagerCluster.update' method (arguments-differ)
```

**Review by ShlomiBalalis — APPROVED** [review]

**Review by fruch — APPROVED** [review]
LGTM

**Comment by karol-kokoszka** [comment]
@ShlomiBalalis @fruch  I need to backport it to manager-3.2 branch. 
Is it enough if I create PR checked out from manager-3.2 branch with a cherry pick of the selected commit from master ?

**Comment by fruch** [comment]
@karol-kokoszka  you marked it for backporting, and now it's backported

**Pull request merged**
